why_postgres.mdx

```diff
@@ -0,0 +1,124 @@
+
+#### SQL Proxy
+
+The key to the solution for most of these issues is our SQL Proxy technology. While there are many Postgres proxies available (e.g. pgbouncer), the Xata SQL Proxy not only speaks the Postgres wire protocol, but also understands the SQL being executed at a fairly deep level. It can, for example, know which tables are accessed, which columns are requests, which functions are being called, with which parameters, and so on. This allows us to block actions that we know would be dangerous for the Postgres cluster. While the security is mostly ensured via Postgres roles, the Proxy can enforce rate and concurrency limits on shared clusters (we don't do that for dedicated clusters).
+
+The Proxy allows us to "virtualize" access not only to the branch, but also to the Postgres catalog of tables, functions, sequences, and so on. ORMs like Prisma and Drizzle do fairly complex introspection to see the tables and columns that are currently defined, and we wanted to make sure they work without any modifications.
+
+You might be wondering why we chose to represent database branches as Postgres schemas rather than Postgres database. The reason has to do with logical replication, which is a key component of the data platform, because reliable change-data-capture (CDC) is such an important feature keeping multiple data stores in sync. Individual databases would have forced us to do a replication slot for each.
+
+#### Logical replication events
+
+Speaking of logical replication, another challenge that we had to solve was getting not only data events but also schema events from Postgres. A well known limitation of the Postgres logical replication system is that it only includes data changes, no DDL changes. Yet, having schema changes in the same stream of events simplifies significantly the downstream consumers for the logical replication events. The way we solve this is by installing [DDL event triggers](https://www.postgresql.org/docs/current/event-triggers.html) that are writing the schema changes to an "out-box" table, which gets replicated via the same replication slot.
+
+We think this tech will be useful for many Postgres applications, so we are planning to open source it soon as its own project.
+
+#### Integrated pgroll
+
+Finally, we have integrated [pgroll]() in our platform. pgroll is our open-source project for zero-downtime, reversable schema migrations. While `pgroll` is CLI tool, in order to integrated it seamlesly in the Xata platform, we have developed a server API for it, which integrates with our control plane. For now, this allows us to keep track of all DDL statements that are being executed and opens the door to a much better schema-changes workflow. Look out for a detailed blog post this week about pgroll and how we integrated it into Xata.
```

Comment:
```suggestion
Finally, we have integrated [pgroll]() in our platform. pgroll is our open-source project for zero-downtime, reversible schema migrations. While `pgroll` is a CLI tool, in order to integrated it seamlessly in the Xata platform, we have developed a server API for it, which integrates with our control plane. For now, this allows us to keep track of all DDL statements that are being executed and opens the door to a much better schema-changes workflow. Look out for a detailed blog post this week about pgroll and how we integrated it into Xata.
```

Reply:
I believe "in order to integrated it seamlessly" should read 
"in order to integrate it seamlessly"